File: test/core/test_ccall_return_pointer.c
#include <emscripten.h>
#include <stdio.h>

int g_data;

EMSCRIPTEN_KEEPALIVE void* getDataPtr() {
  return &g_data;
}

int main() {
  EM_ASM({
    var ptr = ccall("getDataPtr", 'pointer');
    assert(ptr > 0);
  });
  printf("ok\n");
}

File: test/core/test_ccall_return_pointer.js
// This test code fills an array using 'begin < end' comparisons into an array range.
addToLibrary({
  fill_array_up_to__deps: ['$ccall'],
  fill_array_up_to__sig: 'vpi',
  fill_array_up_to: (end_ptr, val) => {
    for(var i = 0;;++i) {
      var ptr = ccall('get_pointer_to_array_index', 'pointer', ['number'], [i]);
      if (ptr >= end_ptr) break;
      {{{ makeSetValue('ptr', '0', 'val', 'i32') }}}
    }
  }
});
